src/index/CompressedRelationHelpersImpl.h

```diff
@@ -0,0 +1,93 @@
+// Copyright 2025 The QLever Authors, in particular:
+//
+// 2021 - 2024 Johannes Kalmbach <kalmbach@cs.uni-freiburg.de>, UFR
+// 2025        Christoph Ullinger <ullingec@informatik.uni-freiburg.de>, UFR
+//
+// UFR = University of Freiburg, Chair of Algorithms and Data Structures
+
+#include "index/CompressedRelation.h"
+
+#ifndef QLEVER_SRC_INDEX_COMPRESSEDRELATIONHELPERSIMPL_H_
+#define QLEVER_SRC_INDEX_COMPRESSEDRELATIONHELPERSIMPL_H_
+
+// Helper function to make a row from `IdTable` easier to compare. This ties
+// the cells of the given row with the indices 0, 1 and 2.
+inline auto tieFirstThreeColumns = [](const auto& row) {
+  return std::tie(row[0], row[1], row[2]);
+};
+
+// Collect elements of type `T` in batches of size 100'000 and apply the
+// `Function` to each batch. For the last batch (which might be smaller)  the
+// function is applied in the destructor.
+CPP_template(typename T, typename Function)(
+    requires ad_utility::InvocableWithExactReturnType<
+        Function, void, std::vector<T>&&>) struct Batcher {
+  Function function_;
+  size_t blocksize_;
+  std::vector<T> vec_;
+
+  Batcher(Function function, size_t blocksize)
+      : function_{std::move(function)}, blocksize_{blocksize} {}
+  void operator()(T t) {
+    vec_.push_back(std::move(t));
+    if (vec_.size() > blocksize_) {
+      function_(std::move(vec_));
+      vec_.clear();
+      vec_.reserve(blocksize_);
+    }
+  }
+  ~Batcher() {
+    if (!vec_.empty()) {
+      function_(std::move(vec_));
+    }
```

Comment:
Can you add a `terminateIfThrows` around this, s.t. in the rare case of exceptions we have at least an idea of what went wrong?